config/config.go

```diff
@@ -270,6 +270,15 @@ func containsTerragruntModule(path string, info os.FileInfo, terragruntOptions *
 		return false, nil
 	}
 
+	// Skip the Terraform data dir
+	dataDir, found := terragruntOptions.Env["TF_DATA_DIR"]
+	if !found {
+		dataDir = options.TerraformDataDir
+	}
```

Comment:
There's a `DataDir` method in `options.go` that does something very similar. Could we extract the logic into one function and use it in both places?

Reply:
I saw this method but it prepends the workspace directory to the data directory path and I wasn't sure if this would work given that we want to ignore any occurrence of the data directory, right? e.g. if a module of a module has a terragrunt.hcl file.

Did you want me to modify that method to conditionally return the workspace + data dir or just the data dir?  Or is my concern unfounded?

Reply:
Looking at the code, and thinking about this a bit more deeply, I think both places may have subtle bugs, and I think creating a common method could fix them.

First, the bugs: 

1. As best as I can tell, `TF_DATA_DIR` can be either a relative or absolute path. If it's a relative path, then the `DataDir()` method below is broken, as it's not prepending that path with the working dir.
1. Similarly, if it's a relative path, then the code here is potentially buggy too. E.g., If `TF_DATA_DIR` is set to `foo`, then the code below would exclude `/foo`, `/foo-bar`, `/.foo`, and a dozen other variations.
1. Moreover, the code below would exclude `/.terraform`, `/.terraform-oops`, `/oops.terraform`, etc.

So I think there are a few things we should do:

1. Create a method called `TerraformDataDir()` that looks up `TF_DATA_DIR` and either returns it or `.terraform` if it's not set.
1. Update the `DataDir()` method to call `TerraformDataDir()` and prepend it with the working dir if it's not an absolute path.
1. Update the code here to (a) call `DataDir()`, (b) call [SplitList](https://golang.org/pkg/path/filepath/#SplitList) on `path`, (c) if `DataDir()` returned an absolute path, call `SplitList` on it too and check if it is an exact prefix of `path`, (d) if `DataDir()` returned a relative path, check if it an exact subsequence of `path`.